lib/internal/modules/esm/resolve.js

```diff
@@ -254,6 +254,9 @@ function finalizeResolution(resolved, base, preserveSymlinks) {
     err.url = String(resolved);
     throw err;
   } else if (!stats.isFile()) {
+    if (process.env.WATCH_REPORT_DEPENDENCIES && process.send) {
```

Comment:
I think we prefix all our environment variables with `NODE_`: https://nodejs.org/api/cli.html#environment-variables

Reply:
Will fix in a follow up PR, as this environment variable existed before this pr